Mollie integration (#3)

mrPauwHaan opened the pull request:
Solves: #29 #58 

Added a Mollie integration for Frappe webforms. Mollie is a big payment provider within Europe and, therefore, an essential for many SMEs who want to use Frappe for handling payments

1. This update integrates Mollie as a payment method in web forms
2. The user is automatically redirected to a checkout page to review the order, and a payment is automatically generated. It is necessary to have a field within your doctype called `payment_id` where the Mollie Payment ID is stored. 
3. The user will be redirected to the Mollie-hosted checkout page, so you can configure any payment method within Mollie and create your own checkout.
4. When the payment is completed, the user is redirected back to the checkout page, where the payment status is automatically verified with Mollie based on the stored payment ID. When the Mollie payment link has expired, is canceled, or has failed, a new payment will automatically be generated on the checkout page, if the payment has not been completed yet.
5. When the payment is completed, the payment url will not be available anymore, and the user will be redirected back to the webform redirect URL.
6. If user is logged in, the email be included as billingadress parameter with the payment. This can overrided by including an `email` field in the form for any custom email. 

Optional
- create a `payment_status` field where the status of the payment is automatically stored.
- Use Mollie in test mode by using the test api key in the mollie settings page. Mollie automatically recognizes it as a test payment.

![image](https://github.com/frappe/payments/assets/120259765/ec0f04ad-d9bb-47a6-aaca-64241f8ad5e8)

![image](https://github.com/frappe/payments/assets/120259765/a44263f3-4f50-453e-a514-f5bf42ebc692)

mrPauwHaan committed:
Mollie integration (#3)

* Update mollie_settings.py

* Update mollie_settings.json

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update stripe_checkout.js

* Update mollie_checkout.js

* Update stripe_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

Daandeve commented:
@mrPauwHaan Great work! But there seems to be an issue with paying, the page keeps loading and I get a console error.
![image](https://github.com/frappe/payments/assets/35433138/13430a6c-a05d-4077-93a9-0ea96fb1e3fb)

Error:
```python
Failed to load resource: the server responded with a status of 500 ()
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 106, in application
    frappe.handler.handle()
  File "apps/frappe/frappe/handler.py", line 49, in handle
    data = execute_cmd(cmd)
  File "apps/frappe/frappe/handler.py", line 85, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "apps/frappe/frappe/__init__.py", line 1697, in call
    return fn(*args, **newargs)
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "apps/payments/payments/templates/pages/mollie_checkout.py", line 69, in make_payment
    paymentID = frappe.db.get_value(reference_doctype, reference_docname, 'payment_id')
  File "apps/frappe/frappe/database/database.py", line 517, in get_value
    result = self.get_values(
  File "apps/frappe/frappe/database/database.py", line 618, in get_values
    out = self._get_values_from_table(
  File "apps/frappe/frappe/database/database.py", line 888, in _get_values_from_table
    return query.run(as_dict=as_dict, debug=debug, update=update, run=run, pluck=pluck)
  File "apps/frappe/frappe/query_builder/utils.py", line 87, in execute_query
    result = frappe.db.sql(query, params, *args, **kwargs)  # nosemgrep
  File "apps/frappe/frappe/database/database.py", line 234, in sql
    self._cursor.execute(query, values)
  File "env/lib/python3.10/site-packages/pymysql/cursors.py", line 153, in execute
    result = self._query(query)
  File "env/lib/python3.10/site-packages/pymysql/cursors.py", line 322, in _query
    conn.query(q)
  File "env/lib/python3.10/site-packages/pymysql/connections.py", line 558, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "env/lib/python3.10/site-packages/pymysql/connections.py", line 822, in _read_query_result
    result.read()
  File "env/lib/python3.10/site-packages/pymysql/connections.py", line 1200, in read
    first_packet = self.connection._read_packet()
  File "env/lib/python3.10/site-packages/pymysql/connections.py", line 772, in _read_packet
    packet.raise_for_error()
  File "env/lib/python3.10/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "env/lib/python3.10/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1054, "Unknown column 'payment_id' in 'field list'")
```

mrPauwHaan commented:
When will this be implemented?

trustedcomputer commented:
@torstenhoegel I don't have any write access to the repository. You'll need to find a frappe maintainer to get this moving.

mrPauwHaan committed:
Merge branch 'frappe:develop' into Ready

mrPauwHaan committed:
Check payment_status field (#4)

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update stripe_checkout.js

* Update mollie_checkout.js

* Update stripe_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

if payment_status exists, first check for status field in doctype before checking the payment ID in the Mollie API

mrPauwHaan committed:
Update mollie_checkout.py

NagariaHussain commented:
Can you attach a video of the working flow?

mrPauwHaan committed:
Update mollie_checkout.py

mrPauwHaan committed:
Update mollie_checkout.py

mrPauwHaan committed:
Update mollie_checkout.js

mrPauwHaan committed:
Update mollie_checkout.js

mrPauwHaan committed:
Update mollie_checkout.py

Only skip if status_field is 'completed'

mrPauwHaan committed:
Update mollie_checkout.js

mrPauwHaan commented:
> Can you attach a video of the working flow?

@NagariaHussain see the video below for the showcase

https://youtu.be/0NwQZI6wNg0

mrPauwHaan commented:
@NagariaHussain @pateljannat @blaggacao when will this be merged? You can find a showcase video above

blaggacao commented:
There's a refactoring going on, taking shape, being validated over at https://github.com/frappe/payments/pull/53

I haven't updated the PR with the lastest additions and bug fixes from production operation.

I think it is worthwhile to hold back on a merge until the new api stabilizes and then refactor.

I can try to push more fixes/additions, I planned to do that anyway, but now that I know there's a stakeholder I might be even more motivated to get this done :smile: 

Thanks for the ping!!

blaggacao commented:
PS: If maintenance is less of an issue for you, for purposes of discoverability, we could add this PR to the readme with a caveat. Users can then deploy from this PR/branch and can report there issues here.

blaggacao commented:
https://github.com/frappe/payments/commit/ef713a20f2e002cf756d45dc6e313162ac1e57ce

timeo-schmidt commented:
This would be important for us too! Any updates on when this will be merged?

mrPauwHaan commented:
@timeo-schmidt As mentioned above, the Payments module is being refactored in #53. When that is stable and the new API is out, this PR needs be be refactored as well and can be merged after.

However, you can still manually use this PR to be able to use the Mollie integration. That is also what I did.

mrPauwHaan commented:
@0spinboson I was recently thinking about working on that, so yes, the intention is there (especially since the Plaid integration in ERPNext no longer works). However, time is my biggest constraint. I am not great with coding, so I wrote this by making many mistakes and spending a lot of time on it. 

Please let me know which features you would like to have. I will consider those if/when I will extend this integration to the payment dashboard

0spinboson commented:
> @0spinboson I was recently thinking about working on that, so yes, the intention is there (especially since the Plaid integration in ERPNext no longer works). However, time is my biggest constraint. I am not great with coding, so I wrote this by making many mistakes and spending a lot of time on it.
> 
> Please let me know which features you would like to have. I will consider those if/when I will extend this integration to the payment dashboard

Well, I recently looked into ERPNext because I'm looking for an alternative to our current bookkeeping/CMS/CRM (and perhaps sales) systems (we're a nonprofit membership association). But I didn't get very far because of the Plaid integration being gone, and I at the very least need to be able to generate invoice/debit batches to send to our bank, if we are to ever switch to this for our membership management and dues administration needs.

In terms of Mollie, we use it both for sales and incoming donations. I'd like a way to get the payment details mollie has listed so I can properly book all the costs and revenues, as well as any/all donations that come in, since there isn't an easy way to get that info from mollie without manual administration.

mahsem commented:
> > @0spinboson I was recently thinking about working on that, so yes, the intention is there (especially since the Plaid integration in ERPNext no longer works). However, time is my biggest constraint. I am not great with coding, so I wrote this by making many mistakes and spending a lot of time on it.
> > Please let me know which features you would like to have. I will consider those if/when I will extend this integration to the payment dashboard
> 
> Well, I recently looked into ERPNext because I'm looking for an alternative to our current bookkeeping/CMS/CRM (and perhaps sales) systems (we're a nonprofit membership association). But I didn't get very far because of the Plaid integration being gone, and I at the very least need to be able to generate invoice/debit batches to send to our bank, if we are to ever switch to this for our membership management and dues administration needs.
> 
> In terms of Mollie, we use it both for sales and incoming donations. I'd like a way to get the payment details mollie has listed so I can properly book all the costs and revenues, as well as any/all donations that come in, since there isn't an easy way to get that info from mollie without manual administration.

@0spinboson  check out this https://www.alyf.de/bank-integration

mrPauwHaan committed:
Update Mollie (#5)

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

if payment_status exists, first check for status field in doctype before checking the payment ID in the Mollie API

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* fix: stripe checkout

* chore: fixed linters

* chore: bumped up semgrep

* fix: stripe page styling

* chore: remove manifest

* chore: add tooling files to gitignore

* ci: adopt best practice

* ci: does it work?

* chore: use ruff, prettier & eslint

* chore: apply pre-commit

* chore: add style commit to blame ignore

* ci: add labeler config

* style: ruff format

* chore: add style commit to blame ignore

* chore: ignore undue lint error

* docs: mention ongoing work

* fix: type conversion #62 (#109)

* refactor: clear API details

* refactor: UI to clear razorpay api details

* chore: translatable label

* refactor: throw for cancelled integration requests

* fix: exception on braintree success page

* fix: get parameters on both success and failed page

* refactor: better handling of get parameters

* refactor: drop pytz

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

* bump mollie api

* Fixed payment page mobile view

* add billing email

---------

Signed-off-by: Akhil Narang <me@akhilnarang.dev>
Co-authored-by: Jannat Patel <pateljannat2308@gmail.com>
Co-authored-by: Jannat Patel <31363128+pateljannat@users.noreply.github.com>
Co-authored-by: David <dgx.arnold@gmail.com>
Co-authored-by: Oswin Dsouza <106068399+Oswin-san@users.noreply.github.com>
Co-authored-by: ruthra kumar <ruthra@erpnext.com>
Co-authored-by: Akhil Narang <me@akhilnarang.dev>

mrPauwHaan committed:
Fixed some errors (#6)

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.html

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_checkout.py

* Update mollie_settings.py

* Update mollie_settings.py

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.html

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

if payment_status exists, first check for status field in doctype before checking the payment ID in the Mollie API

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.py

* Update mollie_checkout.js

* Update mollie_checkout.js

* Update mollie_checkout.py

* Update mollie_checkout.js

* fix: stripe checkout

* chore: fixed linters

* chore: bumped up semgrep

* fix: stripe page styling

* chore: remove manifest

* chore: add tooling files to gitignore

* ci: adopt best practice

* ci: does it work?

* chore: use ruff, prettier & eslint

* chore: apply pre-commit

* chore: add style commit to blame ignore

* ci: add labeler config

* style: ruff format

* chore: add style commit to blame ignore

* chore: ignore undue lint error

* docs: mention ongoing work

* fix: type conversion #62 (#109)

* refactor: clear API details

* refactor: UI to clear razorpay api details

* chore: translatable label

* refactor: throw for cancelled integration requests

* fix: exception on braintree success page

* fix: get parameters on both success and failed page

* refactor: better handling of get parameters

* refactor: drop pytz

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

* bump mollie api

* Fixed payment page mobile view

* add billing email

* added email verification

* fail protection

---------

Signed-off-by: Akhil Narang <me@akhilnarang.dev>
Co-authored-by: Jannat Patel <pateljannat2308@gmail.com>
Co-authored-by: Jannat Patel <31363128+pateljannat@users.noreply.github.com>
Co-authored-by: David <dgx.arnold@gmail.com>
Co-authored-by: Oswin Dsouza <106068399+Oswin-san@users.noreply.github.com>
Co-authored-by: ruthra kumar <ruthra@erpnext.com>
Co-authored-by: Akhil Narang <me@akhilnarang.dev>
